Source/WebKit/NetworkProcess/webrtc/NetworkMDNSRegister.cpp

```diff
@@ -47,9 +48,63 @@ namespace WebKit {
 NetworkMDNSRegister::NetworkMDNSRegister(NetworkConnectionToWebProcess& connection)
     : m_connection(connection)
 {
+#if USE(GLIB)
+    m_cancellable = adoptGRef(g_cancellable_new());
+    g_dbus_proxy_new_for_bus(G_BUS_TYPE_SYSTEM, G_DBUS_PROXY_FLAGS_NONE, nullptr, "org.freedesktop.Avahi", "/", "org.freedesktop.Avahi.Server", m_cancellable.get(), [](GObject*, GAsyncResult* result, gpointer userData) {
+        GUniqueOutPtr<GError> error;
+        auto self = reinterpret_cast<NetworkMDNSRegister*>(userData);
+        self->m_dbusProxy = adoptGRef(g_dbus_proxy_new_for_bus_finish(result, &error.outPtr()));
+        if (error) {
+            LOG_ERROR("Unable to connect to the Avahi daemon: %s", error->message);
+            return;
+        }
```

Comment:
In all the async calls throughout this pull request (and, generally, anywhere), you don't want to print the error message on cancellation:

```suggestion
        if (error) {
                if (!g_error_matches(error.get(), G_IO_ERROR, G_IO_ERROR_CANCELLED))
                    LOG_ERROR("Unable to connect to the Avahi daemon: %s", error->message);
            return;
        }
```

Reply:
Oh right, I think you mentioned it previously and I forgot about it... Thanks for the review, I'll try to update this soon.